pkg/gather/collect/collect.go

```diff
@@ -0,0 +1,489 @@
+		if err != nil {
+			return fmt.Errorf("can't print object: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (c *Collector) writeResource(ctx context.Context, obj kubeinterfaces.ObjectInterface, resourceInfo *ResourceInfo) error {
+	resourceDir := c.getResourceDir(obj, resourceInfo)
+
+	err := os.MkdirAll(resourceDir, 0770)
+	if err != nil {
+		return fmt.Errorf("can't create resource dir %q: %w", resourceDir, err)
+	}
+
+	err = c.writeObject(ctx, resourceDir, obj, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Collector) collect(
+	ctx context.Context,
+	obj kubeinterfaces.ObjectInterface,
+	resourceInfo *ResourceInfo,
+) error {
+	err := c.writeResource(ctx, obj, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func isPublicSecretKey(key string) bool {
+	switch key {
+	case "ca.crt", "tls.crt", "service-ca.crt":
+		return true
+	default:
+		return false
+	}
+}
+
+func (c *Collector) collectSecret(ctx context.Context, u *unstructured.Unstructured, resourceInfo *ResourceInfo) error {
+	secret := &corev1.Secret{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(u.Object, secret)
+	if err != nil {
+		return fmt.Errorf("can't convert secret from unstructured: %w", err)
+	}
+
+	for k := range secret.Data {
+		if !isPublicSecretKey(k) {
+			secret.Data[k] = []byte("<redacted>")
+		}
+	}
+
+	err = c.writeResource(ctx, secret, resourceInfo)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterface, destinationPath string, podName string, logOptions *corev1.PodLogOptions) error {
+	dest, err := os.OpenFile(destinationPath, os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		return fmt.Errorf("can't open file %q: %w", destinationPath, err)
+	}
+	defer func() {
+		err := dest.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close file", "Path", destinationPath)
+		}
+	}()
+
+	logsReq := podClient.GetLogs(podName, logOptions)
+	readCloser, err := logsReq.Stream(ctx)
+	if err != nil {
+		return fmt.Errorf("can't create a log stream: %w", err)
+	}
+	defer func() {
+		err := readCloser.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close log stream", "Path", destinationPath, "Pod", podName, "Container", logOptions.Container)
+		}
+	}()
+
+	_, err = io.Copy(dest, readCloser)
+	if err != nil {
+		return fmt.Errorf("can't read logs: %w", err)
+	}
+
+	return nil
+}
+
+func findContainerStatus(containerStatuses []corev1.ContainerStatus, containerName string) *corev1.ContainerStatus {
+	var s *corev1.ContainerStatus
+	for i := range containerStatuses {
+		if containerStatuses[i].Name == containerName {
+			s = &containerStatuses[i]
+			break
+		}
+	}
+	return s
+}
+
+func (c *Collector) collectContainerLogs(ctx context.Context, logsDir string, pod *corev1.Pod, container *corev1.Container) error {
+	var err error
+	cs := findContainerStatus(pod.Status.ContainerStatuses, container.Name)
```

Comment:
this could use `helpers.Find` and findContainerStatus could be removed

Reply:
ok, I can use `Find` and fix it along the way (it can't enforce comparable)

Reply:
Iirc I've also added FindItem recently with comparable trait

Reply:
the Find signature got fixed post rebase
re `FindItem` - this isn't identity search

Reply:
fixed